### PR TITLE
[Enhancement] fixup isNullable of cast

### PR DESCRIPTION
**Pull request by silverbullet233**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

currently, isNullable always returns true for cast expr, which may not make sense. actually, if target type can be fully compitable with from type, we can keep the nullable property same as child operator.
## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Comment by silverbullet233** [comment]
run starrocks_admit_test

**Comment by silverbullet233** [comment]
run starrocks_admit_test

**Comment by silverbullet233** [comment]
run starrocks_admit_test

**Comment by silverbullet233** [comment]
run starrocks_fe_unittest

**Review by satanson — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/analysis/CastExpr.java [hunk 1]
```diff
@@ -229,6 +229,10 @@ public boolean canHashPartition() {
 
     @Override
     public boolean isNullable() {
+        Expr fromExpr = getChild(0);
+        if (ScalarType.isImplicitlyCastable(fromExpr.getType(), getType(), true)) {
```
[the remark this answers is not in the document]
Comment (silverbullet233) [reply fe/fe-core/src/main/java/com/starrocks/analysis/CastExpr.java:233]: > for the decimal type, could we ensure there won't be null result？

I think we can, null values are only produced when target value can't fully represent source value without loss of precision, if this happens , isImplicitlyCastable must return false

**Review by kangkaisen — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by murphyatwork — DISMISSED** [review]

**Comment by silverbullet233** [comment]
run starrocks_admit_test

**Comment by mofeiatwork** [comment]
run starrocks_admit_test

**Review by murphyatwork — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Pull request merged**